src/Voron/Data/Tables/NewPageAllocator.cs

```diff
@@ -66,18 +67,35 @@ public void Create()
 
         private unsafe Page AllocateMoreSpace(FixedSizeTree fst)
         {
-            var allocatePage = _llt.AllocatePage(NumberOfPagesInSection);
+            int numberOfPagesToAllocate;
+            if (_llt.Environment.Options.ForceUsing32BitsPager || PlatformDetails.Is32Bits)
+                numberOfPagesToAllocate = BitmapSize;             // 256 KB
+            else
+                numberOfPagesToAllocate = NumberOfPagesInSection; // 2 MB
```

Comment:
Why not to set `NumberOfPagesInSection` accordingly instead? Note that have multiple usages of `NumberOfPagesInSection` there (e.g. `FreePage()`) - isn't this a problem?

Reply:
The issue is that we need to keep the same structure between 32 bits & 64 bits systems.
So we can't change the size, because then we wouldn't be able to open the files between platforms.
That is why I'm just marking some of the data as busy upfront.
And in FreePage, the check is merely to catch bad data, which is fine

Reply:
Oh, right.  It's ok then.

Minor issue - let's extract this code to a method and use it in `GetNumberOfPreAllocatedFreePages()` to avoid code duplication